Snackbar: new positions and animation support

steal4life opened the pull request:
Closes #1451

stsrki commented on Source/Extensions/Blazorise.Snackbar/Snackbar.razor.cs:
```diff
@@ -65,7 +69,8 @@ public partial class Snackbar : BaseComponent, IDisposable
     {
         builder.Append( "snackbar" );
-        builder.Append( "snackbar-show", Visible );
+        builder.Append( Visible && !closeAnimation ? "snackbar-show" : null );
```
```suggestion
        builder.Append( "snackbar-show", Visible && !closeAnimation );
```

stsrki commented on Source/Extensions/Blazorise.Snackbar/Snackbar.razor.cs:
```diff
@@ -82,6 +87,13 @@ protected override void OnInitialized()
 
+        if ( closeAnimationTimer == null )
+        {
+            closeAnimationTimer = new( 500 );
```
Add parameter CloseAnimationInterval so that it isn't hardcoded at 500.

stsrki commented on Source/Extensions/Blazorise.Snackbar/Snackbar.razor.cs:
```diff
@@ -130,12 +139,23 @@ public Task Show()
+    async Task ProposeHide( SnackbarCloseReason closeReason )
+    {
+        closeAnimation = true;
+        closeAnimationTimer.Delay( 500 );
```
Add parameter `CloseAnimationInterval` so that it isn't hardcoded at 500.

stsrki commented on Source/Extensions/Blazorise.Snackbar/Snackbar.razor.cs:
```diff
@@ -130,12 +139,23 @@ public Task Show()
         return InvokeAsync( StateHasChanged );
     }
 
+    async Task ProposeHide( SnackbarCloseReason closeReason )
```
Rename `ProposeHide` to something like `BeginHide`, or `ExecuteHide`.

stsrki commented on Source/Extensions/Blazorise.Snackbar/Utils/Extensions.cs:
```diff
@@ -20,8 +20,9 @@ public static string GetName( this SnackbarLocation snackbarLocation )
+                return "bottom";
             default:
-                return null;
+                return "";
```
Always return null instead of empty string.

stsrki commented on Source/Extensions/Blazorise.Snackbar/Utils/Extensions.cs:
```diff
@@ -43,8 +44,9 @@ public static string GetName( this SnackbarStackLocation snackbarStackLocation )
+                return "bottom";
             default:
-                return null;
+                return "";
```
Always return null instead of empty string.

David-Moreira commented on Documentation/Blazorise.Docs/Pages/Docs/Extensions/Snackbar/SnackbarPage.razor:
```diff
@@ -104,9 +104,12 @@
         Allow snackbar to show multiple lines of text.
     </DocsAttributesItem>
-    <DocsAttributesItem Name="DefaultInterval" Type="double" Default="5000">
+    <DocsAttributesItem Name="Interval" Type="double" Default="5000">
```
Are we changing the naming of an existing **Parameter**?

stsrki replied:
This was wrong in the docs. It was always named `Interval`.

David-Moreira commented on Demos/Blazorise.Demo/Pages/Tests/SnackbarsPage.razor:
```diff
@@ -157,15 +177,19 @@
-@code {
+
+ @code {
     Snackbar snackbar;
     Snackbar snackbarAction;
     Snackbar snackbarMultiline;
     Snackbar snackbarLeft;
     Snackbar snackbarRight;
+    Snackbar snackbarTop;
+    Snackbar snackbarTopEnd;
+    Snackbar snackbarTopStart;
```
Could almost just have a configurable location parameter?

stsrki replied:
Tried it. Makes it complicated without reason.

David-Moreira commented on Source/Extensions/Blazorise.Snackbar/Enums/SnackbarLocation.cs:
```diff
@@ -0,0 +1,59 @@
+    Top,
+
+    /// <summary>
+    /// Show the snackbar on the top-left side of the screen.
```
the documentation refers to left & right. it should refer to start or end. we should not assume the left or right position when naming Start/End.

steal4life replied:
So you think it should be Top-Start?

David-Moreira replied:
I personally do. In the future we plan to have RTL support. So in that case the doc would be incorrect.

David-Moreira commented on Source/Extensions/Blazorise.Snackbar/Snackbar.razor.cs:
```diff
@@ -44,7 +46,12 @@ public partial class Snackbar : BaseComponent, IDisposable
-    /// Flag that indicates if snackbar close action was delayed.
+    /// A flag that indicates if the snackbar is currently being closed with animation.
+    /// </summary>
+    private bool closeAnimation = false;
```
Would `closingAnimation` be a better naming then? It would be consistent with `closingDelayed` and the documentation we've added also talks about "being closed" => closing

steal4life replied:
Yes, definitely

David-Moreira commented on Source/Extensions/Blazorise.Snackbar/Utils/Extensions.cs:
```diff
@@ -6,11 +6,21 @@ public static string GetName( this SnackbarLocation snackbarLocation )
+            case SnackbarLocation.Top:
+                return "top";
+            case SnackbarLocation.TopStart:
+                return "top-left";
+            case SnackbarLocation.TopEnd:
+                return "top-right";
```
Can we have start/end support? Or does it not make sense here?

steal4life replied:
if i understand, you think that placing the snackbar in the middle of the screen on the left or right side is what can be added?... If so, i don't think it makes sense because the snackbar animations work by showing it from the bottom or top of the screen, possibly add an animation like offcanvas to slide on left or right side of the screen

stsrki replied:
Agree, no need to make it at center left/right.

David-Moreira replied:
I was asking about the css class naming. If it needs to be left right instead of start end

David-Moreira commented on Source/Extensions/Blazorise.Snackbar/wwwroot/blazorise.snackbar.css:
```diff
@@ -1,7 +1,7 @@
   align-items: center;
   background-color: var(--b-snackbar-background, #323232);
-  color: var(--b-snackbar-text-color, white);
+  color: var(--b-snackbar-text-color, rgb(255, 255, 255));
```
curious, any special reason for this apparent redudant change?

stsrki replied:
@steal4life ?

David-Moreira commented:
1. The **Top-Right-Aligned** takes a bit too long to show. Also if we want to show animations maybe it should be more explicit? A configuration like it's on the other pages? That just sets the animation for all the snackbars in the page? So the user or us can try it for every single one.
http://localhost:7713/tests/snackbar
2. Does the positioning also work correctly with stacked snackbars? That would be great.
3. Update enum doc:
![image](https://github.com/Megabit/Blazorise/assets/22283161/86aa1564-cc72-4e89-99f4-2c3f1f005264)

steal4life commented:
> 1. The **Top-Right-Aligned** takes a bit too long to show. Also if we want to show animations maybe it should be more explicit? A configuration like it's on the other pages? That just sets the animation for all the snackbars in the page? So the user or us can try it for every single one.
>    http://localhost:7713/tests/snackbar
> 2. Does the positioning also work correctly with stacked snackbars? That would be great.
> 3. Update enum doc:
>    ![image](https://user-images.githubusercontent.com/22283161/258581132-86aa1564-cc72-4e89-99f4-2c3f1f005264.png)

So the Top right appears longer due to the added parameter
AnimationDuration, which determines the length of the snackbar animation (how long it will last), with this we have enabled users to determine the length of the snackbar animation themselves in their future projects if they want so.

Yes, we added that the stack can also have the same positions as the snackbar.

David-Moreira commented:
> > 1. The **Top-Right-Aligned** takes a bit too long to show. Also if we want to show animations maybe it should be more explicit? A configuration like it's on the other pages? That just sets the animation for all the snackbars in the page? So the user or us can try it for every single one.
> >    http://localhost:7713/tests/snackbar
> > 2. Does the positioning also work correctly with stacked snackbars? That would be great.
> > 3. Update enum doc:
> >    ![image](https://user-images.githubusercontent.com/22283161/258581132-86aa1564-cc72-4e89-99f4-2c3f1f005264.png)
> 
> So the Top right appears longer due to the added parameter AnimationDuration, which determines the length of the snackbar animation (how long it will last), with this we have enabled users to determine the length of the snackbar animation themselves in their future projects if they want so.
> 
> Yes, we added that the stack can also have the same positions as the snackbar.

But does it make sense for 1 to be configurable in the page? Helps testing and seeing every use case.

steal4life commented:
I just added that parameter on one element to see the diference bettwen
that one and others, but we can do as you sad so it will make more sense.

On Sat, 5 Aug 2023, 16:37 David Moreira, ***@***.***> wrote:

>
>    1. The *Top-Right-Aligned* takes a bit too long to show. Also if we
>    want to show animations maybe it should be more explicit? A configuration
>    like it's on the other pages? That just sets the animation for all the
>    snackbars in the page? So the user or us can try it for every single one.
>    http://localhost:7713/tests/snackbar
>    2. Does the positioning also work correctly with stacked snackbars?
>    That would be great.
>    3. Update enum doc:
>    [image: image]
>    <https://user-images.githubusercontent.com/22283161/258581132-86aa1564-cc72-4e89-99f4-2c3f1f005264.png>
>
> So the Top right appears longer due to the added parameter
> AnimationDuration, which determines the length of the snackbar animation
> (how long it will last), with this we have enabled users to determine the
> length of the snackbar animation themselves in their future projects if
> they want so.
>
> Yes, we added that the stack can also have the same positions as the
> snackbar.
>
> But does it make sense for 1 to be configurable in the page? Helps testing
> and seeing every use case.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Megabit/Blazorise/pull/4900#issuecomment-1666522954>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS23XEJA4SVAJBXLYXV2MLDXTZLB7ANCNFSM6AAAAAA23HMRL4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

steal4life commented:
Yes, tehnicaly in scss there is for instance snackbar-top-right etc... So
if we want to change the name here we should also do the same in scss.

On Sat, 5 Aug 2023, 16:33 David Moreira, ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In Source/Extensions/Blazorise.Snackbar/Utils/Extensions.cs
> <https://github.com/Megabit/Blazorise/pull/4900#discussion_r1285067586>:
>
> > +                return "top-left";
> +            case SnackbarLocation.TopEnd:
> +                return "top-right";
>
> I was asking about the css class naming. If it needs to be left right
> instead of start end
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Megabit/Blazorise/pull/4900#discussion_r1285067586>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS23XEPMD76ZU3U3SLJS3DDXTZKUVANCNFSM6AAAAAA23HMRL4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

David-Moreira commented on Documentation/Blazorise.Docs/Pages/Docs/Helpers/Enums/SnackbarPage.razor:
```diff
@@ -9,9 +9,15 @@
-            <UnorderedListItem><Code>Default</Code> Default behavior.</UnorderedListItem>
-            <UnorderedListItem><Code>Left</Code> Show the snackbar on the left side of the screen.</UnorderedListItem>
-            <UnorderedListItem><Code>Right</Code> Show the snackbar on the right side of the screen.</UnorderedListItem>
+            <UnorderedListItem><Code>Default</Code> Default behavior (obsolete, use <Code>Bottom</Code> instead).</UnorderedListItem>
+            <UnorderedListItem><Code>Bottom</Code> Show the snackbar on the bottom side of the screen.</UnorderedListItem>
+            <UnorderedListItem><Code>BottomStart</Code> Show the snackbar on the bottom-left side of the screen.</UnorderedListItem>
+            <UnorderedListItem><Code>BottomEnd</Code> Show the snackbar on the bottom-right side of the screen.</UnorderedListItem>
+            <UnorderedListItem><Code>Start</Code> Show the snackbar on the left side of the screen (obsolete, use <Code>BottomStart</Code> instead).</UnorderedListItem>
+            <UnorderedListItem><Code>End</Code> Show the snackbar on the right side of the screen (obsolete, use <Code>BottomEnd</Code> instead).</UnorderedListItem>
+            <UnorderedListItem><Code>Top</Code> Show the snackbar on the top side of the screen.</UnorderedListItem>
+            <UnorderedListItem><Code>TopStart</Code> Show the snackbar on the top-left side of the screen.</UnorderedListItem>
+            <UnorderedListItem><Code>TopEnd</Code> Show the snackbar on the top-right side of the screen.</UnorderedListItem>
```
left/right vs start/end

David-Moreira commented:
In demo, Multiline snackbar and Info snackbar don't seem to be working with animation, either the parameter was forgotten or something's up

steal4life commented:
Problably accidentaly, I did not notice it before.

On Sun, 6 Aug 2023, 12:49 Mladen Macanović, ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In Source/Extensions/Blazorise.Snackbar/wwwroot/blazorise.snackbar.css
> <https://github.com/Megabit/Blazorise/pull/4900#discussion_r1285194759>:
>
> > @@ -1,7 +1,7 @@
>  ﻿.snackbar {
>    align-items: center;
>    background-color: var(--b-snackbar-background, #323232);
> -  color: var(--b-snackbar-text-color, white);
> +  color: var(--b-snackbar-text-color, rgb(255, 255, 255));
>
> @steal4life <https://github.com/steal4life> ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Megabit/Blazorise/pull/4900#discussion_r1285194759>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS23XEMK43JRXVBYTAK5UFLXT5ZDBANCNFSM6AAAAAA23HMRL4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
